app/helpers/patient_query_helper.rb

```diff
@@ -2,37 +2,42 @@
 
 # Helper methods for filtering through patients
 module PatientQueryHelper # rubocop:todo Metrics/ModuleLength
-  def patients_table_data(params)
+  def patients_table_data(params, current_user)
     # Require workflow and tab params
     workflow = params.require(:query).require(:workflow).to_sym
     tab = params.require(:query).require(:tab).to_sym
 
-    # Validate filter and sorting params
-    begin
-      query = validate_patients_query(params.require(:query))
-    rescue StandardError => e
-      return render json: e, status: :bad_request
-    end
+    query = validate_patients_query(params.require(:query))
 
     # Validate pagination params
     entries = params.require(:query)[:entries]&.to_i || 25
+    raise InvalidQueryError.new(:entries, entries) unless entries >= 0
+
     page = params.require(:query)[:page]&.to_i || 0
-    return render json: { error: 'Invalid entries or page' }, status: :bad_request unless entries >= 0 && page >= 0
+    raise InvalidQueryError.new(:page, page) unless page >= 0
 
     # Get filtered patients
     patients = patients_by_query(current_user, query)
 
+    # Filter out current monitoree if exclude_patient_id is defined
+    exclude_patient_id = query[:exclude_patient_id]&.to_i || nil
+    raise InvalidQueryError.new(:exclude_patient_id, exclude_patient_id) unless exclude_patient_id.nil? || (exclude_patient_id >= 0)
```

Comment:
minor suggestion to avoid double negative

```suggestion
    raise InvalidQueryError.new(:exclude_patient_id, exclude_patient_id) if exclude_patient_id.present? && exclude_patient_id < 0
```